pkg/skaffold/inspect/buildEnv/add_gcb.go

```diff
@@ -71,6 +71,57 @@ func AddGcbBuildEnv(ctx context.Context, out io.Writer, opts inspect.Options) er
 	return inspect.MarshalConfigSet(cfgs)
 }
 
+func ModifyGcbBuildEnv(ctx context.Context, out io.Writer, opts inspect.Options) error {
+	formatter := inspect.OutputFormatter(out, opts.OutFormat)
+	cfgs, err := inspect.ConfigSetFunc(config.SkaffoldOptions{ConfigurationFile: opts.Filename, ConfigurationFilter: opts.Modules, SkipConfigDefaults: true, MakePathsAbsolute: util.BoolPtr(false)})
+	if err != nil {
+		return formatter.WriteErr(err)
+	}
+	if opts.Profile == "" {
+		// empty profile flag implies that only modify the default pipelines in the target `skaffold.yaml`
+		cfgs = cfgs.SelectRootConfigs()
+		for _, cfg := range cfgs {
+			if cfg.Build.GoogleCloudBuild == nil {
+				if opts.Strict {
+					return formatter.WriteErr(inspect.BuildEnvNotFound(inspect.BuildEnvs.GoogleCloudBuild, cfg.SourceFile, ""))
+				}
+				continue
+			}
+			cfg.Build.GoogleCloudBuild = constructGcbDefinition(cfg.Build.GoogleCloudBuild, opts.BuildEnvOptions)
+			cfg.Build.LocalBuild = nil
+			cfg.Build.Cluster = nil
+		}
+	} else {
+		profileFound := false
+		for _, cfg := range cfgs {
+			index := -1
+			for i := range cfg.Profiles {
+				if cfg.Profiles[i].Name == opts.Profile {
+					index = i
+					break
+				}
+			}
+			if index < 0 {
+				continue
+			}
+			profileFound = true
+			if cfg.Profiles[index].Build.GoogleCloudBuild == nil {
+				if opts.Strict {
+					return formatter.WriteErr(inspect.BuildEnvNotFound(inspect.BuildEnvs.GoogleCloudBuild, cfg.SourceFile, opts.Profile))
+				}
+				continue
+			}
+			cfg.Profiles[index].Build.GoogleCloudBuild = constructGcbDefinition(cfg.Profiles[index].Build.GoogleCloudBuild, opts.BuildEnvOptions)
+			cfg.Profiles[index].Build.LocalBuild = nil
+			cfg.Profiles[index].Build.Cluster = nil
```

Comment:
I was envisioning the given profile to be already a GCB profile and this command will error out if its not a GCB profile.  From the implementation looks like, it will force the profile build env to be GCB and override `local` or `cluster`

Reply:
sorry could you take another look, I think it's doing what you're expecting.